docs/language_specification/general_overview.md

```diff
@@ -20,18 +18,30 @@ A cQASM program can contain the following types of tokens:
 
 ## Statements
 
-A cQASM program consists of a sequence of *statements*.
-Among those, a quantum algorithm is in essence a sequence of *instructions*:
-*gates* and non-unitary quantum operations applied to qubit arguments
-(_e.g._, the [*measure instruction*](measure_instruction.md)).
+A cQASM program consists of a list of *statements*, more concretely,
+a *version statement* followed by *variable declarations* and *instructions*. 
+In essence, a quantum algorithm is expressed via a sequence of unitary and non-unitary quantum operations
+applied to qubit arguments
+(_e.g._, the [*measure instruction*](statements/instructions/non_unitary_instructions/measure_instruction.md)).
+The unitary operations can be either *named gates* or compositions of *gate modifiers* acting on a named gate.
```

Comment:
Added the `commonly known as gates`, but not added the link, because we would be pointing from `unitary operations` to `named gates`, which is just a subset of them.